Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/HapticsProvider.kt

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package org.dolphinemu.dolphinemu.utils
+
+import android.os.Build
+import android.os.VibrationEffect
+import android.os.Vibrator
+import androidx.annotation.FloatRange
+import androidx.annotation.RequiresApi
+import org.dolphinemu.dolphinemu.features.input.model.DolphinVibratorManagerFactory
+
+/**
+ * This class provides methods that facilitate performing haptic feedback.
```

Comment:
Abstract/vague language. How about something more direct like:

```suggestion
 * Sends haptic feedback to the user.
```

Reply:
If simple descriptions are preferred, then does the following sound right for a class?

> Provides haptic feedback to user

Reply:
Sure, that sounds fine. And the description doesn't necessarily have to be short - I just think the description as it is contain a lot of words that don't add any meaning.